src/optimisationAlgorithm/covarianceMatrixAdaptationEvolutionStrategy.cpp

```diff
@@ -0,0 +1,359 @@
+#include "mantella_bits/optimisationAlgorithm/covarianceMatrixAdaptationEvolutionStrategy.hpp"
+
+// C++ standard library
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <limits>
+
+// Armadillo
+#include <armadillo>
+
+// Mantella
+#include "mantella_bits/armadillo.hpp"
+#include "mantella_bits/assert.hpp"
+#include "mantella_bits/optimisationProblem.hpp"
+
+namespace mant {
+  CovarianceMatrixAdaptationEvolutionStrategy::CovarianceMatrixAdaptationEvolutionStrategy()
+      : OptimisationAlgorithm(),
+        lambda_(std::numeric_limits<arma::uword>::max()) {
+    setIsStagnatingFunction(
+        [this](
+            const arma::Mat<double>& parameters_,
+            const arma::Row<double>& objectiveValues_,
+            const arma::Row<double>& differences_) {
+          assert(objectiveValues_.n_elem == parameters_.n_cols);
+          assert(differences_.n_elem == parameters_.n_cols);
+          assert(differences_.has_inf() || arma::all(objectiveValues_ - differences_ - arma::min(objectiveValues_) + arma::min(differences_) < 1e-12 * arma::max(arma::ones<arma::Row<double>>(arma::size(objectiveValues_)), arma::abs(objectiveValues_))));
+          
+          //encountered a hard error, like eigenwert < 0
+          if(sigma_ <= 0) {
+            return true;
+          }
+          if (arma::max(diagD_) > 1e14 * arma::min(diagD_)) {
```

Comment:
Mhm, mein Vorschlag:
//1e14 is a magic threshold decided by Hansen to limit the condition by